src/trajectory.jl

```diff
@@ -326,31 +348,30 @@ abstract type DynamicTrajectory{I<:AbstractIntegrator} <: AbstractTrajectory{I}
 
 """
 $(TYPEDEF)
-
 Standard HMC implementation with fixed total trajectory length.
 
 # Fields
-
 $(TYPEDFIELDS)
 
 # References
 1. Neal, R. M. (2011). MCMC using Hamiltonian dynamics. Handbook of Markov chain Monte Carlo, 2(11), 2. ([arXiv](https://arxiv.org/pdf/1206.1901)) 
 """
-struct HMCDA{S<:AbstractTrajectorySampler,I<:AbstractIntegrator} <: DynamicTrajectory{I}
-    "Integrator used to simulate trajectory."
-    integrator  ::  I
-    "Total length of the trajectory, i.e. take `floor(λ / integrator_step)` number of leapfrog steps."
-    λ           ::  AbstractFloat
+struct HMCDA{S<:AbstractTrajectorySampler, I<:AbstractIntegrator, TC<:StaticTerminationCriterion} <: DynamicTrajectory{I}
```

Comment:
Maybe `HMCDA` should be a subtype of `StaticTrajectory{I}`?

Reply:
`StaticTrajectory` is not yet an abstract type in this PR. This will be updated when unifying the types, which is planned after this PR.